libsolutil/JSON.h

```diff
@@ -33,12 +33,31 @@ namespace solidity::util
 /// Removes members with null value recursively from (@a _json).
 Json::Value removeNullMembers(Json::Value _json);
 
+/// JSON printing format.
+struct JsonPrintingFormat
+{
+	/// JSON format.
+	enum class Format { Compact, Pretty } format = Format::Compact;
+
+	/// Default indent level.
+	static constexpr uint32_t defaultIndent = 2;
+
+	/// Indent level when Format::Pretty is used.
+	uint32_t indent = defaultIndent;
+
+	bool operator==(JsonPrintingFormat const& _other) const noexcept;
+	bool operator!=(JsonPrintingFormat const& _other) const noexcept { return !(*this == _other); }
```

Comment:
- I think it would be better to group fields together. At first I did not even notice that `format` was a separate field.
- Also, we usually put fields last.
- If you make it a normal `enum` (rather than `enum class`), the names will become much shorter but still readable. E.g `JsonFormat::Pretty`.
- Most of the comments state obvious things so I'd drop them.

```suggestion
	enum Format
	{
		Compact,
		Pretty,
	};

	static constexpr uint32_t defaultIndent = 2;

	bool operator==(JsonPrintingFormat const& _other) const noexcept;
	bool operator!=(JsonPrintingFormat const& _other) const noexcept { return !(*this == _other); }

	Format format = Compact;
	uint32_t indent = defaultIndent;
```